nvflare/client/in_process/api.py

```diff
@@ -0,0 +1,203 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import time
+from typing import Any, Dict, Optional, Union
+
+from nvflare.apis.analytix import AnalyticsDataType
+from nvflare.apis.fl_constant import FLMetaKey
+from nvflare.apis.shareable import Shareable
+from nvflare.app_common.abstract.fl_model import FLModel, ParamsType
+from nvflare.app_common.utils.fl_model_utils import FLModelUtils
+from nvflare.client.api_spec import APISpec
+from nvflare.client.config import ClientConfig, ConfigKey, from_file
+from nvflare.client.constants import SYS_ATTRS
+from nvflare.client.utils import DIFF_FUNCS
+from nvflare.fuel.data_event.data_bus import DataBus
+from nvflare.fuel.data_event.event_manager import EventManager
+
+TOPIC_LOG_DATA = "LOG_DATA"
+TOPIC_STOP = "STOP"
+TOPIC_ABORT = "ABORT"
+TOPIC_LOCAL_RESULT = "LOCAL_RESULT"
+TOPIC_GLOBAL_RESULT = "GLOBAL_RESULT"
+
+
+class InProcessClientAPI(APISpec):
```

Comment:
can we add docstring to this class as well to explain each input argument for constructor?